File: CS490/src/program1/node.java
package program1;

/**
 * Class for instantiating a node object
 *
 * @author  Cameron Howell
 * @version 1.0
 * @since   2020-10-01
 */
public class node {
    private int pid;
    private int priority;
    private int time_ms;

    /**
     * Default constructor
     */
    node() {
        System.out.println("Node created.");
    }

    /**
     * Constructor
     * @param pid process id
     * @param priority priority level for execution order determination
     * @param time_ms time slice in milliseconds
     */
    node(int pid, int priority, int time_ms) {
        this.pid = pid;
        this.priority = priority;
        this.time_ms = time_ms;
    }

    /**
     * Accessor for pid
     * @return the current pid
     */
    int getPid() {
        return this.pid;
    }

    /**
     * Accessor for priority
     * @return the current priority
     */
    int getPriority() {
        return this.priority;
    }

    /**
     * Accessor for time_ms
     * @return the current time slice
     */
    int getTime_ms() {
        return this.time_ms;
    }
}

File: CS490/Program2/Program2/Node.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Program2
{
    /// <summary>
    /// Node class for instantiating and working with a node,
    /// or "process"
    /// </summary>
    public class Node
    {
        private readonly int pid;
        private readonly int priority;
        private readonly int time_ms;

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="id">ID number for the process</param>
        /// <param name="pri">The process priority level</param>
        /// <param name="t">Time in milliseconds</param>
        public Node(int id, int pri, int t)
        {
            pid = id;
            priority = pri;
            time_ms = t;
        }

        /// <summary>
        /// Accessor for the pid
        /// </summary>
        /// <returns>The node's process id</returns>
        public int GetPid()
        {
            return pid;
        }

        /// <summary>
        /// Accessor for the priority
        /// </summary>
        /// <returns>The node's priority level</returns>
        public int GetPriority()
        {
            return priority;
        }

        /// <summary>
        /// Accessor for the time in milliseconds
        /// </summary>
        /// <returns>The node's time in ms</returns>
        public int GetTimeMs()
        {
            return time_ms;
        }
    }
}
